src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/SchemaGeneratorOptions.cs

```diff
@@ -9,15 +9,15 @@ public class SchemaGeneratorOptions
     {
         public SchemaGeneratorOptions()
         {
-            CustomTypeMappings = new Dictionary<Type, Func<OpenApiSchema>>();
+            CustomTypeMappings = new Dictionary<Type, Func<MappingContext, OpenApiSchema>>();
             SchemaIdSelector = DefaultSchemaIdSelector;
             SubTypesSelector = DefaultSubTypesSelector;
             DiscriminatorNameSelector = DefaultDiscriminatorNameSelector;
             DiscriminatorValueSelector = DefaultDiscriminatorValueSelector;
             SchemaFilters = new List<ISchemaFilter>();
         }
 
-        public IDictionary<Type, Func<OpenApiSchema>> CustomTypeMappings { get; set; }
+        public IDictionary<Type, Func<MappingContext, OpenApiSchema>> CustomTypeMappings { get; set; }
```

Comment:
This is a binary-breaking change, so if we took it as-is this would have to be part of v7.

Reply:
This would get released sooner if there were a way to implement this without breaking changes.